HermaFx.Foundation/RuntimeExtensions/StreamExtensions.cs

```diff
@@ -68,5 +69,30 @@ public static byte[] ReadAllBytes(this Stream source)
 		{
 			return ReadAllBytes(source, 0);
 		}
+
+		/// <summary>
+		/// Copy the contents of a stream to a destination.
+		/// thrown if it is cancelled during the process.
+		/// </summary>
+		/// <param name="source">The source.</param>
+		/// <exception cref="OperationCanceledException">Occurs in a thread when an operation being executed by that thread is cancelled.</exception>
+		public static void CopyTo(this Stream @this, Stream destination, int bufferSize, CancellationToken cancellationToken)
+		{
+			var buffer = new byte[bufferSize];
+			int count;
+
+			using (var @out = new MemoryStream())
+			{
+				while ((count = @this.Read(buffer, 0, buffer.Length)) != 0)	
+				{
+					if (cancellationToken.IsCancellationRequested)
+						throw new OperationCanceledException(string.Format("{0}: Operation cancelled... aborting request!", nameof(CopyTo)));
+
+					@out.Write(buffer, 0, count);
```

Comment:
Seems like you don't need this new stream as you are copying stuff into the another one